test/e2e/framework/util.go

```diff
@@ -65,6 +64,44 @@ func WaitForServiceAccount(ctx context.Context, c corev1client.CoreV1Interface,
 	return event.Object.(*corev1.ServiceAccount), nil
 }
 
+func WaitForServiceAccountTokenSecret(ctx context.Context, c corev1client.CoreV1Interface, namespace, name string) (*corev1.Secret, error) {
+	fieldSelector := fields.OneTermEqualSelector("metadata.name", name).String()
+	lw := &cache.ListWatch{
+		ListFunc: helpers.UncachedListFunc(func(options metav1.ListOptions) (runtime.Object, error) {
+			options.FieldSelector = fieldSelector
+			return c.Secrets(namespace).List(ctx, options)
+		}),
+		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+			options.FieldSelector = fieldSelector
+			return c.Secrets(namespace).Watch(ctx, options)
+		},
+	}
+	event, err := watchtools.UntilWithSync(
+		ctx,
+		lw,
+		&corev1.Secret{},
+		nil,
+		func(e watch.Event) (bool, error) {
+			switch t := e.Type; t {
+			case watch.Added, watch.Modified:
+				s := e.Object.(*corev1.Secret)
+				return len(s.Data[corev1.ServiceAccountTokenKey]) > 0, nil
```

Comment:
this could panic

Reply:
Would there be any other reason than the interface conversion? I'm looking at existing code (https://github.com/scylladb/scylla-operator/blob/7e9ea2c21ee76299f5342419c37513a70eab556c/test/e2e/framework/util.go#L50) almost identical to this and I'm wondering whether this was missed before or if I'm missing something now?

Reply:
the cast is fine but maps can be nil

Reply:
Fetching from nil map would still return a zero value though?

Reply:
you are right, i though it will panic but that's only for writes